src/audio.h

```diff
@@ -0,0 +1,58 @@
+/*
```

Comment:
How difficult would it be to split out this stuff to a folder with implementations? Similar to the `system` folder? Have the thread handled internally as well since some libraries (for example port audio) handle threads already (or uses signals on some backends).